lib/models/association/BaseAssociation.ts

```diff
@@ -8,8 +8,8 @@ import {ModelClassGetter} from '../../types/ModelClassGetter';
 import {ModelNotInitializedError} from '../errors/ModelNotInitializedError';
```

Comment:
I suggest reverting all changes here and move all repository mode related stuff to https://github.com/RobinBuschmann/sequelize-typescript/blob/master/lib/models/BaseSequelize.ts#L104 Then only the sequelize instance need to care about repository stuff.

What do you think?